server/util.js

```diff
@@ -28,6 +28,9 @@ export const genGiftCard = giftCardId => {
 export const authGiftCard = async jwtGiftCard => {
   let giftCardId = jwt.verify(jwtGiftCard, process.env.JWT_GIFT_SECRET)
   let giftCard = await GiftCard.findById(giftCardId)
+  if (giftCard === null) {
```

Comment:
if(giftCard){
   var elapsedTime = new Date...
}
else throw new Error('This gift card not available!')